src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -821,7 +832,45 @@ private static Task ProcessUnknownFrameAsync()
             await OnEndStreamReceived();
         }
 
-        if (!_isMethodConnect && (_parsedPseudoHeaderFields & _mandatoryRequestPseudoHeaderFields) != _mandatoryRequestPseudoHeaderFields)
+        if (_isProtocolWebTransport)
+        {
+            // Requirement in the spec but this is handled by the Http/3 layer already, so leaving commented out for posterity
```

Comment:
I left this commented out to avoid confusion for those who will be reviewing and maintaining the code later on so they can more easily see all the checks enforced by the spec. However, I know some teams don't like checking in commented code. Should I remove this?

Reply:
Go ahead and remove this, it's not part of the CONNECT spec, only the WebTransport spec.